### PR TITLE
Feature/suspended twitter accounts

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Closes https://github.com/DemocracyClub/yournextrepresentative/issues/978

This work covers two functions related to suspended Twitter accounts: 

1. When adding/updating a twitter username that has been suspended, the username will still be valid. 
2. When Twitterbot checks usernames on a routine basis, it will no longer delete suspended accounts, but rather log a suspended message and save the suspended status to `extra_data`. 

We've setup two separate API calls for Twitterbot to capture suspended accounts. The reason is due to the limitations on the Twitter API endpoints:

- ..`lookup.json `can handle a request for multiple usernames
- ..`show.json` can return suspended accounts
- Neither can do both.

 TO DO
- [x] Fix tests
- [x] Extract suspended twitter account logic
- [x] Add test for suspended accounts
- [ ] Add screenshots

### CODE REVIEW

File: ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py [hunk 1]
```diff
@@ -54,10 +55,21 @@ def handle_person(self, person):
...
                                 f"{person}'s Twitter account (id: {user_id}) is currently suspended."
                             )
-                        # def LoggedAction to show that account was suspended
+                            if (
+                                identifier.extra_data
+                                == "Twitter account (id: {user_id}) is currently suspended"
+                            ):
+                                continue
+                            else:
+                                identifier.extra_data += "Twitter account (id: {user_id}) is currently suspended"
```
Comment (VirginiaDooley) [inline ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:67]: Note to self: store as a dict

File: ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py [hunk 2]
```diff
@@ -54,10 +55,21 @@ def handle_person(self, person):
...
+                                identifier.save()
+                                LoggedAction.objects.create(
```
Comment (VirginiaDooley) [inline ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:69]: note to self: Record fk for person

Comment (VirginiaDooley) [reply ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:69]: note to self: Record source as Twitterbot

File: ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py [hunk 3]
```diff
@@ -54,10 +55,21 @@ def handle_person(self, person):
...
+                                    action_type="suspended-twitter-account"
+                                )
```
Comment (VirginiaDooley) [inline ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:71]: note to self: add action to 

And create a ticket to refactor that file

File: ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py [hunk 4]
```diff
@@ -54,10 +55,21 @@ def handle_person(self, person):
 
                     if data:
                         if any(d["code"] == 63 for d in data["errors"]):
-                            return verbose(
+                            verbose(
```
Comment (VirginiaDooley) [inline ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:58]: note to self: extract 44-79 into a new method and pass in a list of user ids and screen names

Comment (VirginiaDooley) [reply ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:58]: @michaeljcollinsuk Should this be extracted to a helper method or a management method?

Comment (VirginiaDooley) [reply ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:58]: @symroe ^^^

Comment (symroe) [reply ynr/apps/twitterbot/management/commands/twitterbot_update_usernames.py:58]: Good question - I think this is best done in the management command, as there's already a lot of logic in here. I do generally like to keep management commands small, with logic in helpers, but in this case let's not start a refactor

**Comment by michaeljcollinsuk** [comment]
I've updated and rebased the PR - ready for a final review now.

On CI fails - one is a very minor coverage failure related to a previously untested method (`get_twitter_user_id`). 

And the CodeQL failure is due to code unrelated to this PR so a) I'm not sure why it has cropped up now and b) I am a bit wary of making any changes as its unrelated

**Pull request merged**